### PR TITLE
Add link and update example for Ed25519 Signature Suite 2018.

### PR DESCRIPTION
**Pull request by peacekeeper**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -146,19 +146,20 @@ <h1>The Registry</h1>
           Ed25519SigningKey
       </td>
       <td>
-          None
+          <a href="https://w3c-dvcg.github.io/lds-ed255192018/">Ed25519 Signature Suite 2018</a>
```
Comment (msporny) [inline index.html]: Very minor nit... all those numbers are hard to parse in the name and may lead to developers mistyping something? or easy to mistake 2018 from 2019? I don't know... wonder if we can make it easier for humans to parse the string... e.g. "EdwardsSignatureSuite2018", "EdwardsKey2018". People are going to be typing the latter string quite a bit... and it may be easier to type and parse "EdwardsKey2018" rather than "Ed255192018"?

Also wonder if we should name the spec lds-edwards2018 or ldp-edwards2018? 

We may be renaming the Linked Data Signatures spec to Linked Data Proofs. Or we may just make Linked Data Signatures a subclass of Linked Data Proofs. Thinking on this isn't super clear right now, so I don't think you need to change the lds/ldp stuff in this PR.

File: index.html [hunk 2]
```diff
@@ -146,19 +146,20 @@ <h1>The Registry</h1>
...
-  "id": "did:example:123456789abcdefghi/keys/2",
-  "type": "Ed25519SigningKey",
-  "owner": "did:example:123456789abcdefghi",
-  "expires": "2017-02-08T16:02:20Z",
-  "publicKeyBase64": "IOmA4R7TfhkYTYW87...CBMq2/gi25s="
+  "@context": ["https://w3id.org/security/v1"],
+    "id": "did:example:123456789abcdefghi#keys-1",
+    "type": "Ed25519VerificationKey",
+    "owner": "did:example:123456789abcdefghi",
+    "publicKeyBase64" : "H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV"
```
Comment (dlongley) [inline index.html]: `publicKeyBase64` is a base64-encoding of what format specifically?

Comment (peacekeeper) [reply index.html]: Ed25519 keys are simply 32 raw bytes that don't have a "format".
But actually I think I made a mistake in this example, the value string is in Base58 encoding, not Base64. So I should either change the value string, or use "publicKeyBase58" instead of "publicKeyBase64"....

Comment (dlongley) [reply index.html]: https://blog.mozilla.org/warner/2011/11/29/ed25519-keys/

Ok, since in this case the data is *literally* the raw public key bytes, I think the name is appropriate (with the base58 fix you mentioned).

Comment (mikelodder7) [reply index.html]: Generally if you just use base64 or hex we should be talking about the raw bytes with no specific structure like ASN.1. Can we make sure that that is clear somewhere in a notes section or referenced somewhere?

**Comment by peacekeeper** [comment]
Okay I think we can use `publicKeyHex` for now. To address @mikelodder7 's point, I added a note to https://w3c-dvcg.github.io/lds-ed255192018/ which explains that this property contains the raw public key bytes.

**Comment by mikelodder7** [comment]
Looks good to me. Thanks for adding that.

**Comment by dlongley** [comment]
@peacekeeper,

JFYI - added a quick implementation here:

https://github.com/digitalbazaar/jsonld-signatures/tree/ed25519

It uses `publicKeyBase58`. Do we want to support hex?

**Comment by dlongley** [comment]
IMO, we'll want to update to use `proof` instead of `signature` and `jws` instead of `signatureValue`. That way we get both "proof" and JWS compatibility.

**Comment by peacekeeper** [comment]
`publicKeyHex` is what the 'btcr' method seems to be using right now. On the other hand Indy/Sovrin natively return Base58, so `publicKeyBase58` is fine with me too. @mikelodder7 is working on the 'sov' method spec, what's your preference?

**Comment by mikelodder7** [comment]
I prefer Base58 for Sovrin so there is no need to change encodings.

**Comment by msporny** [comment]
> I prefer Base58 for Sovrin

There are three encoding schemes for base58... which one is being used here?

https://en.wikipedia.org/wiki/Base58

Also note that there is no normative spec for base58, so we'll have to point to /something/ normative-ish. No idea what that spec/doc is.

**Comment by peacekeeper** [comment]
For Sovrin it's the same as for Bitcoin and IPFS. Can we just 'declare' this to be used for the `publicKeyBase58` property, or should we use a more specific property name? (Or should we use `publicKeyHex` to avoid uncertainty?)

**Comment by mikelodder7** [comment]
There are no three different encodings for Base58. There is Base58 and Base58Check which includes CRC bytes at the end. Sovrin uses Base58Check. I will double check that. The property can be publicKeyBase58Check. See https://en.bitcoin.it/wiki/Base58Check_encoding

**Comment by dlongley** [comment]
@mikelodder7,

> There are no three different encodings for Base58.

Per the link, there are actually three different alphabets used for Base58. One that Bitcoin and IPFS use, one that Ripple uses, and one that Flickr uses. Base58Check is yet another format, which uses the Bitcoin/IPFS alphabet, includes CRC bytes at the end as you said, and also includes a version/application byte at the beginning.

If we're going to support `publicKeyBase58Check` we'll need to figure out what to do about that version/application byte. It may be cleaner to just use Base58 with the Bitcoin/IPFS alphabet. But I'm not strongly opposed to supporting `publicKeyBase58Check` if we can do something that makes sense and won't confuse other applications -- which may mean simply coming up with our own version/application byte. But it seems to lose its meaning (and simply becomes overhead) when that version/application byte is the same regardless of application -- which is how we'd use it.

**Comment by mikelodder7** [comment]
I personally am not seeing any difference in the letters used in those three applications. They all use the same alphabet and exclude the same characters and just have different hashes. That being said, the more I think about it, I think would should just opt for Base58. Base58Check was created to help prevent errors for users as they typed their private keys into the computer. I don't think we are planning to have users type in DIDs (maybe we will, but I haven't heard any discussion on this.) I think Base58 should be preferred unless there is a specific reason to choose Base58Check. I agree with you @dlongley that we should just use the Bitcoin/IPFS format.

**Comment by dlongley** [comment]
@mikelodder7,

> I personally am not seeing any difference in the letters used in those three applications. They all use the same alphabet and exclude the same characters and just have different hashes.

Jfyi - the alphabets are in a different order. So `0x00` maps to the char `1` in the Bitcoin alphabet but it maps to the char `r` in the Ripple alphabet.

```js
const bs58 = require('bs58');
const rbs58 = require('ripple-bs58');

const b1 = new Buffer([0, 1]);

console.log('bitcoin', bs58.encode(b1));
console.log('ripple', rbs58.encode(b1));
```

Output:
```
bitcoin 12
ripple rp
```

> I think Base58 should be preferred unless there is a specific reason to choose Base58Check.

+1

**Comment by mikelodder7** [comment]
Thanks @dlongley that helps to clarify it.

**Comment by peacekeeper** [comment]
Okay so should we go with `publicKeyBase58` with the Bitcoin/IPFS alphabet for now? (In this case, the PR should be fine the way it is). I can add another note to the [lds-ed255192018](https://w3c-dvcg.github.io/lds-ed255192018/) doc to explain this.

**Comment by mikelodder7** [comment]
+1

**Comment by msporny** [comment]
Can we pick a better name for the suite and keys? I suggest:

Name of the suite: "EdwardsSignatureSuite2018"
Name of keys used by the suite: "EdwardsKey2018"
Name of the spec: "lds-edwards2018"

I suggest the specific form we use for '2018 is ed25519. Thoughts?

**Review by msporny — CHANGES_REQUESTED** [review]
Minor nits, let's please change.

File: index.html (hunk 1 above)
Comment (msporny) [reply index.html]: Sorry, just noticed that these comments I made a few days ago weren't submitted.

Comment (dlongley) [reply index.html]: Developers would see/type: `Ed25519Signature2018` (so the curve numbers are separated from the year). I also think most developers would be looking for `ed25519` not `edwards`.

The spec name should change though. I agree with that -- too hard to read.

Comment (msporny) [reply index.html]: Ok, so:

* Change spec name to lds-ed25519-2018 (and maybe we should change all the spec repo names to match this format)
* Keep the name as Ed25519Signature2018

Right? If so, we have consensus and the PR should be merged?

**Review by msporny — APPROVED** [review]
LGTM, merging.
